src/gui/activitydata.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) by Klaas Freitag <freitag@owncloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#ifndef ACTIVITYDATA_H
+#define ACTIVITYDATA_H
+
+#include <QtCore>
+
+namespace OCC {
+/**
+ * @brief The ActivityLink class describes actions of an activity
+ *
+ * These are part of notifications which are mapped into activities.
+ */
+
+class ActivityLink
+{
+public:
+    QString _label;
+    QString _link;
+    QString _verb;
+    bool _isPrimary;
+};
+
+/* ==================================================================== */
+/**
+ * @brief Activity Structure
+ * @ingroup gui
+ *
+ * contains all the information describing a single activity.
+ */
+
+class Activity
+{
+public:
+    typedef QPair<qlonglong, QString> Identifier;
+
+    enum Type {
+        ActivityType,
+        NotificationType
+    };
+
+    Type      _type;
+    qlonglong _id;
+    QString   _subject;
+    QString   _message;
+    QString   _file;
+    QUrl      _link;
+    QDateTime _dateTime;
+    QString   _accName;
+
+    QVector <ActivityLink> _links;
+    /**
+     * @brief Sort operator to sort the list youngest first.
+     * @param val
+     * @return
+     */
+    bool operator<( const Activity& val ) const;
```

Comment:
nitpick:  in general it is better to have operator< and operator==  as non member function (for symetry reason)